src/pages/index.js

```diff
@@ -5,96 +5,235 @@ import PopupWithImage from "../components/PopupWithImage.js";
+const deleteCardForm = document.forms["delete-card-form"];
+const profileFormElement = document.forms["profile-edit-fields"];
+
+//Button consts
+const profileAddButton = document.querySelector(".profile__add-button");
+const profileEditButton = document.querySelector(".profile__edit-button");
+const editAvatarButton = document.querySelector(".profile__avatar-container");
 
 //Initializing cards
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
 
-cardSection.renderItems();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "a5df8bb7-ccf7-4af1-a820-819df810a6c4",
+    "Content-Type": "application/json",
+  },
+});
 
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([cards, userInfo]) => {
+    console.log(cards, userInfo);
+  })
+  .catch((err) => console.error(err));
+
+let cardSection;
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (data) => {
+          const cardElement = createCard(data);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+//initializing user info
 const userInfo = new UserInfo({
   nameSelector: "#profile-title",
   descriptionSelector: "#profile-job",
+  avatarImage: "#profile-image",
 });
 
-//Initializing add place popup window
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo(userData.name, userData.about);
+  userInfo.setUserAvatar(userData.avatar);
+});
+
+//Initializing popup windows
 const addPlacePopup = new PopupWithForm(
   "#add-place-form",
   handleAddCardFormSubmit
 );
 
-addPlacePopup.setEventListeners();
-
-//Initializing add place popup window
+//Initializing profile popup window
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-window",
   handleProfileFormSubmit
 );
-profileEditPopup.setEventListeners();
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar",
+  handleAvatarFormSubmit
+);
 
 const cardPreview = new PopupWithImage("#image-popup", handleImageClick);
 
-//Edit profile window consts
-const profileFormElement = document.forms["profile-edit-fields"];
+const deletePopup = new DeletePopup("#delete-card", deleteCardForm);
+
+//Set eventlisteners
+
+addPlacePopup.setEventListeners();
+profileEditPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
+deletePopup.setEventListeners();
+
+//Form Validaors
 
 const editFormValidator = new FormValidator(config, profileFormElement);
 editFormValidator.enableValidation();
 
-//New card window consts
-const addNewPlaceForm = document.forms["add-place-edit-fields"];
-
 const addFormValidator = new FormValidator(config, addNewPlaceForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```